src/components/MiniSearchUI.js

```diff
@@ -7,28 +7,62 @@ import NavbarSearch from '../components/NavbarSearch';
 import SearchResults from '../components/SearchResults';
 import Scrollable from '../components/Scrollable';
 import SearchResultsCount from '../components/SearchResultsCount';
+import ObjectUtils from '../util/ObjectUtils';
+import QueryResponse from '../api/QueryResponse';
 
 type MiniSearchUIProps = {
   /**
    * A scale (1.0 = 100%) to use when rendering the search results part of
    * the MiniSearchUI. Optional—defaults to 100%.
    */
   scale: number;
+  /**
+   * Optional callback to handle searching if not using the context's
+   * Searcher component to do the searching (otherwise the searcher's onSearch method is used).
+   */
+  onSearch: null | (q: string) => void;
+  /**
+   * Optional callback to handle updating the query string if not using the context's
+   * Searcher component to do the searching (otherwise the searcher's onSearch method is used).
+   */
+  updateQuery: null | (q: string) => void;
+  /**
+   * Optional query response to render if not using the context's
+   * Searcher component to do the searching (otherwise the searcher's state's response is used).
+   */
+  response: QueryResponse | null;
+  /**
+   * Optional query error to render if not using the context's
+   * Searcher component to do the searching (otherwise the searcher's state's error is used).
+   */
+  error: string | null;
 };
 
 type MiniSearchUIDefaultProps = {
   scale: number;
+  onSearch: null | (q: string) => void;
```

Comment:
Unlike `MiniSearchUIProps`, `MiniSearchUIDefaultProps` is explicitly defined in your component. defaultProps are based on your specifications on lines `60-66` and are not based on what the parent passes through as a prop. Keeping that in mind, `onSearch`, `updateQuery`, `response`, and `error`, can only be `null` here.

Reply:
Done. Made all defaultProps null, removed `onSearch` and `updateQuery` from defaultProps.